common/djangoapps/enrollment/views.py

```diff
@@ -609,20 +609,28 @@ def post(self, request):
                 )
             else:
                 # Will reactivate inactive enrollments.
-                response = api.add_enrollment(username, unicode(course_id), mode=mode, is_active=is_active)
+                enrollment_detail = api.add_enrollment(username, unicode(course_id), mode=mode, is_active=is_active)
 
             email_opt_in = request.data.get('email_opt_in', None)
             if email_opt_in is not None:
                 org = course_id.org
                 update_email_opt_in(request.user, org, email_opt_in)
 
-            return Response(response)
+            return Response(
+                status=status.HTTP_200_OK,
+                data={
+                    "message": (
+                        u"The user {username}' has been enrolled in course '{course_id}."
+                    ).format(username=username, course_id=course_id),
+                    "enrollment_detail": enrollment_detail
+                }
+            )
         except CourseModeNotFoundError as error:
             return Response(
                 status=status.HTTP_400_BAD_REQUEST,
                 data={
                     "message": (
-                        u"The course mode '{mode}' is not available for course '{course_id}'."
+                        u"The course mode '{mode}' is either not available or expired for course '{course_id}'."
```

Comment:
`The [{mode}] course mode is expired or otherwise unavailable for course run [{course_id}].`